src/fork_gc.c

```diff
@@ -1334,6 +1334,8 @@ static int periodicCb(void *privdata) {
     gc->retryInterval.tv_sec = RSGlobalConfig.gcConfigParams.forkGc.forkGcRetryInterval;
     IndexSpecRef_Release(early_check);
     RedisModule_ThreadSafeContextUnlock(ctx);
+    close(gc->pipe_read_fd);
+    close(gc->pipe_write_fd);
     return 1;
   }
```

Comment:
Can we just move this check to before we create the pipe?
Does this check require the GIL?
@lerman25

Reply:
Yes
`// GIL must be held before calling this function`
`static inline bool isOutOfMemory(RedisModuleCtx *ctx) {`